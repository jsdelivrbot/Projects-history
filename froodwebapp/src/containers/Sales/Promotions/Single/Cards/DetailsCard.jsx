import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Row, Col } from 'react-flexbox-grid';
import { Card, Button } from 'antd';
import { Field } from 'redux-form';
import {
  FormSelect,
  FormInput,
  FormInputNumber,
  FormCheckbox
} from 'components';
import { getMenuItems } from 'utils';
import styles from '../Promotion.scss';

const mapStateToProps = state => ({
  selectValues: state.commonData.promotionFields
});

const DetailsCard = ({
  selectValues,
  type,
  setMinPurchage,
  hasMinPurchase,
  handleFormatter,
  handleParser,
  // selectedSku,
  handleToggleModal,
  constants
}) => (
  <Col className={ styles.cardWrapper }>
    <Card className={ styles.card }>
      <Row middle="xs">
        <Col xs={ 12 } md={ 4 } lg={ 4 }>Promotion Type</Col>
        <Col xs={ 12 } md={ 5 } lg={ 5 }>
          <Field
            name="type"
            component={ FormSelect }
            props={ {
              menuItems: getMenuItems(selectValues.discountType) || []
            } }
          />
        </Col>
      </Row>
      { Number(type) === constants.freeItem &&
        <Row start="xs">
          <Col xs={ 12 } md={ 4 } lg={ 4 } />
          <Col xs={ 12 } md={ 5 } lg={ 5 }>
            <Button id="type" onClick={ handleToggleModal }>Add SKU</Button>
          </Col>
        </Row>
      }
      { Number(type) !== constants.freeShipping &&
        <Row middle="xs">
          <Col xs={ 12 } md={ 4 } lg={ 4 } />
          <Col xs={ 12 } md={ 5 } lg={ 5 }>
            <Field
              name="value"
              placeholder={ type !== constants.freeItem ? 'Number' : 'Value' }
              step={ type === constants.valueOff ? 0.1 : 1 }
              component={ type !== constants.freeItem ? FormInputNumber : FormInput }
              formatter={ handleFormatter }
              parser={ handleParser }
              disabled={ type === constants.freeItem }
            />
          </Col>
        </Row>
      }
      { Number(type) === constants.freeItem &&
        <Row start="xs">
          <Col xs={ 12 } md={ 4 } lg={ 4 } >Quantity</Col>
          <Col xs={ 12 } md={ 5 } lg={ 5 }>
            <Field
              name="qty"
              placeholder="Quantity"
              component={ FormInputNumber }
            />
          </Col>
        </Row>
      }
      <Row middle="xs" className={ styles.floatLeft }>
        <Col md={ 4 } lg={ 4 } />
        <Col xs={ 12 } md={ 5 } lg={ 5 }>
          <Field
            name="requireMinumumPurchase"
            text="Required minimum purchase"
            onChange={ setMinPurchage }
            component={ FormCheckbox }
          />
        </Col>
      </Row>
      { hasMinPurchase &&
        <Row middle="xs">
          <Col xs={ 12 } md={ 4 } lg={ 4 } >
            Minimum purchase
          </Col>
          <Col xs={ 12 } md={ 5 } lg={ 5 }>
            <Field
              name="minumumPurchase"
              step={ 0.1 }
              placeholder={ type !== 2 ? 'Number' : 'Value' }
              component={ type !== 2 ? FormInputNumber : FormInput }
            />
          </Col>
        </Row>
      }
    </Card>
  </Col>
);

DetailsCard.propTypes = {
  selectValues: PropTypes.object.isRequired,
  type: PropTypes.number,
  setMinPurchage: PropTypes.func,
  handleFormatter: PropTypes.func,
  handleParser: PropTypes.func,
  handleToggleModal: PropTypes.func,
  hasMinPurchase: PropTypes.bool,
  constants: PropTypes.object,
};

export default connect(mapStateToProps)(DetailsCard);
