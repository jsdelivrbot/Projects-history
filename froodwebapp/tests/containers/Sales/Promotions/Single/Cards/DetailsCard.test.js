import DetailsCard from 'containers/Sales/Promotions/Single/Cards/DetailsCard';
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

const { describe, expect, it } = global;

describe('DetailsCard', () => {
  const props = {
    selectValues: [],
    type: 1,
    condition: 1,
    selectedSku: 'sku',
    setMinPurchage: jest.fn(),
    handleFormatter: jest.fn(),
    handleParser: jest.fn(),
    handleToggleModal: jest.fn(),
    hasMinPurchase: false,
    constants: {}
  };

  it('renders the DetailsCard component', () => {
    const component = shallow(<DetailsCard { ...props } />);

    expect(component.length).toEqual(1);
  });

  it('capturing Snapshot of DetailsCard', () => {
    const component = shallow(<DetailsCard { ...props } />);
    const view = toJson(component);
    const actual = expect(view);

    actual.toMatchSnapshot();
  });
});
