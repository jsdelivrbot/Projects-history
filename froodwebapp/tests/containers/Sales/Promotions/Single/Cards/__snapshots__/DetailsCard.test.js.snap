// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsCard capturing Snapshot of DetailsCard 1`] = `
<Col
  className="cardWrapper"
>
  <Card
    className="card"
  >
    <Row
      middle="xs"
    >
      <Col
        lg={4}
        md={4}
        xs={12}
      >
        Promotion Type
      </Col>
      <Col
        lg={5}
        md={5}
        xs={12}
      >
        <Field
          component={[Function]}
          name="type"
          props={
            Object {
              "menuItems": Array [],
            }
          }
        />
      </Col>
    </Row>
    <Row
      middle="xs"
    >
      <Col
        lg={4}
        md={4}
        xs={12}
      />
      <Col
        lg={5}
        md={5}
        xs={12}
      >
        <Field
          component={[Function]}
          disabled={false}
          formatter={[Function]}
          name="value"
          parser={[Function]}
          placeholder="Number"
          step={1}
        />
      </Col>
    </Row>
    <Row
      className="floatLeft"
      middle="xs"
    >
      <Col
        lg={4}
        md={4}
      />
      <Col
        lg={5}
        md={5}
        xs={12}
      >
        <Field
          component={[Function]}
          name="requireMinumumPurchase"
          onChange={[Function]}
          text="Required minimum purchase"
        />
      </Col>
    </Row>
  </Card>
</Col>
`;
